inc/class-tmsc-linked-taxonomy-posts.php

```diff
@@ -0,0 +1,358 @@
+	 * @param boolean $update.
+	 * @return object. WP_Term object.
+	 */
+	public function save_linked_taxonomy( $post_id, $post = null, $update = true ) {
+		// Short circut the write of data if needed.
+		// This is useful when you need to map multiple taxonomies to a single post type as the realtionship must be 1 to 1.
+		if ( apply_filters( 'tmsc_skip_linked_taxonomy_save', false ) ) {
+			return;
+		}
+
+		$term = $this->get_linked_term( $post_id );
+
+		if ( empty( $term ) ) {
+
+
+			$post = ( empty( $post ) ) ? get_post( $post_id ) : $post;
+			$post_type = ( 'revision' === $post->post_type ) ? get_post_type( $post->post_parent ) : $post->post_type;
+
+			if ( ! empty( $this->linked_types[ $post_type ] ) ) {
+				// If we want to link a post to terms instead of taxonomies, we calculate the parent terms here.
+				if ( is_array( $this->linked_types[ $post_type ] ) ) {
+					$taxonomy = $this->get_query_arg( 'taxonomy' );
+					if ( empty( $taxonomy ) || ! in_array( $taxonomy, $this->linked_types[ $post_type ], true ) ) {
+						return;
+					}
+				} elseif ( is_string( $this->linked_types[ $post_type ] ) ) {
+					$taxonomy = $this->linked_types[ $post_type ];
+				} else {
+					// No need to save anything.
+					return;
+				}
+				// Don't save our initial auto draft.
+				if ( 'auto-draft' !== $post->post_status ) {
+					// Often the top level posts will act as a container for child posts and we won't need terms for these.
+					// Use the filter hook||config below to enable it for child posts.
+					if ( empty( $post->post_parent ) || apply_filters( 'tmsc_link_child_posts', false, $post, $taxonomy ) ) {
+
+						// If filter hook above has been applied, then write a child term.
+						$parent_term = 0;
+						if ( ! empty( $this->parent_term ) ) {
+							$parent_term = $this->parent_term->term_id;
+						}
+
+						$args = array(
+							'slug' => $post->post_name,
+							'parent' => ( ! empty( $parent_term ) ) ? $parent_term : 0,
+						);
+						$term = wp_insert_term( $post->post_title, $taxonomy, $args );
+
+						if ( ! empty( $term ) && ! is_wp_error( $term ) ) {
+							update_term_meta( $term['term_id'], $this->linked_term_meta_key, $post_id );
+							update_post_meta( $post_id, $this->linked_post_meta_key, $term['term_id'] );
+							update_post_meta( $post_id, $this->linked_taxonomy_meta_key, $taxonomy );
+							$term = get_term( $term['term_id'], $taxonomy );
+						}
+					} else {
+						$term = $this->get_linked_term( $post->post_parent );
+						// Set the meta data for the children.
+						update_term_meta( $term->term_id, $this->linked_term_meta_key, $post->post_parent );
+						update_post_meta( $post_id, $this->linked_post_meta_key, $term->term_id );
+						update_post_meta( $post_id, $this->linked_taxonomy_meta_key, $taxonomy );
+					}
+				}
+			}
+		} // End if().
+
+		$this->clear_linked_cache( $post_id );
+		if ( ! empty( $term ) && ! is_wp_error( $term ) ) {
+			return $term;
```

Comment:
Why do you need to return the term object here?  This function is hooked into the save_post action which does not need any return value.